src/openrct2/localisation/Localisation.cpp

```diff
@@ -1223,39 +1223,59 @@ void format_string_to_upper(utf8 *dest, size_t size, rct_string_id format, void
     }
 }
 
-money32 string_to_money(char * string_to_monetise)
+money32 string_to_money(const char* string_to_monetise)
 {
     const char* decimal_char = language_get_string(STR_LOCALE_DECIMAL_POINT);
-    char * text_ptr = string_to_monetise;
-    int i, j, sign;
+    char* text_ptr = const_cast<char *>(string_to_monetise);
+    sint32 sign;
+    uint32 numNumbers = 0, numMinus = 0, numDecSeps = 0;
     // Remove everything except numbers decimal, and minus sign(s)
-    for (i = 0; text_ptr[i] != '\0'; ++i) {
+    for (uint32 i = 0, j; text_ptr[i] != '\0'; ++i)
+    {
+        if (text_ptr[i] >= '0' && text_ptr[i] <= '9')
+            numNumbers++;
+        if (text_ptr[i] == decimal_char[0])
+            numDecSeps++;
+        if (text_ptr[i] == '-')
```

Comment:
I think the minus sign should only be legal when in first position (as in `[0]`). No need to salvage otherwise.

Incidentally, in what context does it make sense for a user to pass a negative monetary value?

Reply:
@AaronVanGeffen `money_to_string` was originally introduced back when I improved on the add/set money option. I wanted it to be able to handle any scenario the user threw at it, even someone setting the money to a negative value for, say, a scenario about a park in debt. 

However, I'm not sure why I didn't make it the first position. I do remember having to hack alot of it together as I was still new to C and remember fighting tooth and nail with transitioning from c# string ideas to a C array.